### PR TITLE
Deleted timetableslots cleanup

### PR DESCRIPTION
**Pull request by flexik**
This implements detection and marking of deleted timetable slots and should solve remaining data errors in #143.

### CODE REVIEW

File: app/interactors/cleanup_removed_timetable_slots.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require 'interpipe/interactor'
+require 'interactors/cleanup_removed_records'
+
+class CleanupRemovedTimetableSlots < CleanupRemovedRecords
+  include Interpipe::Interactor
+
+  def perform(timetable_slots: , faculty_semester: , **options)
+    @slots = timetable_slots
+    @options = options
+    stored_slot_ids = TimetableSlot.join(Parallel, id: :parallel_id) \
```
Comment (jirutka) [inline app/interactors/cleanup_removed_timetable_slots.rb]: Is it really needed to end this line with escape character?

File: app/interactors/cleanup_removed_timetable_slots.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    .where(semester: faculty_semester.code, faculty: faculty_semester.faculty, timetable_slots__deleted_at: nil)
+    .select(Sequel.lit('timetable_slots.*'))
+    .select_map(:timetable_slots__id)
```
Comment (jirutka) [inline app/interactors/cleanup_removed_timetable_slots.rb]: These 3 lines should be indented…

Comment (flexik) [inline app/interactors/cleanup_removed_timetable_slots.rb]: Fixed

Comment (jirutka) [inline app/interactors/cleanup_removed_timetable_slots.rb]: It doesn’t look like so…

**Comment by jirutka** [comment]
I don’t like subclassing from `CleanupRemovedRecords`. It’s IMO unnecessary, `cleanup_records` looks more like an utility method. However, I don’t consider this as blocking for merge.

**Comment by jirutka** [comment]
:information_source: This is now deployed on staging.

**Pull request merged**